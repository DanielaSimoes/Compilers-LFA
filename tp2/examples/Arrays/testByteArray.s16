program testArray ()
{
    byte a[] = {'a','b'};
    byte n[] = {0x1,0x2};
    byte c[] = {0x1,0x2,'c'};
    byte d[] = {0x1,1,'c'};
    int t[] = {1,2,3};
    int x[] = {0x1,0x2};
    exit();
}
