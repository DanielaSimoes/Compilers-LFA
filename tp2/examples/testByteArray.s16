program testArray ()
{
    byte n[] = {'1','2'};
    exit();
}
