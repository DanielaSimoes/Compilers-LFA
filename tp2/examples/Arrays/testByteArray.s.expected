               ; creating an array
               .data
a:             .byte  a, b
               ; creating an array
n:             .byte  1, 2
               ; creating an array
c:             .byte  1, 2, c
               ; creating an array
d:             .byte  1, 1, c
               ; creating an array
t:             .word  1, 2, 3
               ; creating an array
x:             .word  1, 2
               ; exiting program
               .text
L0:
               halt

               ; includes
               halt
printInt:
               ipush 28
printInt1:     dup2
               iushr
               ipush 0xF
               iand
               jsr pNibble
               dup
               ifeq printInt2
               ipush 4
               isub
               goto printInt1
printInt2:     pop
               ret

pNibble:
               ipush 0x30
               iadd
               dup
               ipush 0x39
               isub
               ifle pNibble1
               ipush 0x0a
               iadd
pNibble1:      write
               ret

